### ISSUE TITLE
Undeclared variable inside a string is missed.

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#2129]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086, SC2154, SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

declare package=$1

print $package
print "$package"
print $pacakge
print "$pacakge"  ### line 8

```

#### Here's what shellcheck currently says:
```
Line 5	SC2086: Double quote to prevent globbing and word splitting.
Line 7	SC2154: pacakge is referenced but not assigned.
Line 7	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

```
Line 5	SC2086: Double quote to prevent globbing and word splitting.
Line 7	SC2154: pacakge is referenced but not assigned.
Line 7	SC2086: Double quote to prevent globbing and word splitting.
Line 8	SC2154: pacakge is referenced but not assigned.
```

**Comment by matthewpersico** [comment]
This bit me **hard** today. Thanks.

**Comment by TinCanTech** [comment]
Something is strange:
```
#!/bin/sh

TEST=0

echo "$TETS"
echo "$TETS"
```
shellcheck.net:
```
$ shellcheck myscript
 
Line 3:
TEST=0
^-- SC2034: TEST appears unused. Verify use (or export if used externally).

$ 
```

**Comment by TinCanTech** [comment]
Followed by:
```
#!/bin/sh

FOO=0

echo "$bah"
echo "$bah"
```
shellcheck.net:
```
$ shellcheck myscript
 
Line 3:
FOO=0
^-- SC2034: FOO appears unused. Verify use (or export if used externally).
 
Line 5:
echo "$bah"
      ^-- SC2154: bah is referenced but not assigned.

$ 
```
Weird ..

**Comment by TinCanTech** [comment]
@matthewpersico I believe `shellcheck` only warns for the first occurrence of an unassigned variable.

**Comment by matthewpersico** [comment]
I just tried this:
```
#!/bin/bash

declare package=$1

print $package
print "$package"
print "$pacakge"  ### line 8
```
and got
```
$ shellcheck myscript
 
Line 5:
print $package
      ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
print "$package"
 
Line 7:
print "$pacakge"  ### line 8
       ^-- SC2154: pacakge is referenced but not assigned.

```
so yes, it's not a quoted issue, it's first error only issue. I must have misspelled it twice then. Is there a mode or a config to make it flag ALL instances, not just he first. In an editor where it's interactive, you can live with it this way: fix one, the next one pops up. Kinda like Whack-a-Bug. I imagine in a CI setting not interactive, that's got to be annoying to have to re-execute to find each one. Maybe change the error:

```
print "$pacakge"  ### line 8
       ^-- SC2154: First instance: pacakge is referenced but not assigned.
```
so at least you've been warned to do a global replace.

**Comment by TinCanTech** [comment]
> so yes, it's not a quoted issue, it's first error only issue. _I must have misspelled it twice then_.

You did:
> print "$package"
> print $**pacakge**
> print "$**pacakge**"  ### line 8

> Is there a mode or a config to make it flag ALL instances, not just he first.

I haven't stumbled over that yet.

> In an editor where it's interactive, you can live with it this way: fix one, the next one pops up. Kinda like Whack-a-Bug. 

:rofl:

**Comment by matthewpersico** [comment]
> > so yes, it's not a quoted issue, it's first error only issue. _I must have misspelled it twice then_.
> 
> You did:

No, no, I meant I must have misspelled it twice in my ORIGINAL code. Here was just for the demo. But funny, I don't remember seeing any shellcheck markers for the first instance. I'm going to go re-break my code to see if I can repro with real code.

**Comment by matthewpersico** [comment]
Yep. Works as expected. First one flagged, subsequent not. In and out of quotes. This is not an issue.
